src/input/input_json.rs

```diff
@@ -144,7 +144,10 @@ impl<'a> Input<'a> for JsonInput {
             JsonInput::Uint(u) => Ok(EitherInt::U64(*u)),
             JsonInput::BigInt(b) => Ok(EitherInt::BigInt(b.clone())),
             JsonInput::Float(f) => float_as_int(self, *f),
-            JsonInput::String(str) => str_as_int(self, str),
+            JsonInput::String(str) => match str_as_int(self, str) {
+                Ok(i) => Ok(i),
+                Err(_) => str_as_int(self, &str.replace('_', "")),
+            },
```

Comment:
The idea is to only eat the performance cost in the error case. This means that parsing `1_000` will be slower than `1000` but I think `1000` is more common so that should be okay.

Reply:
To avoid duplication should we put this inside `str_as_int` ? It also has `strip_decimal_zeros` path, we could combine the two stripping operations for efficiency's sake.